lib/core/types.ts

```diff
@@ -8,6 +8,7 @@ export interface MainOptions extends Options {
   exportTypeName: string;
   exportTypeInterface: string;
   listDifferent: boolean;
+  header?: string;
```

Comment:
Even though it's optional via the CLI, I believe this value always exists so we shouldn't need to mark it as optional.

```suggestion
  header: string;
```

Reply:
it's true that the value is always set when called from the CLI code, but I think making it required is a worse internal API.  the header really is optional, allowing callers to omit the argument creates no ambiguity.  making it required complicates the code; not in a huge way, but various tests would need updating with an extra line of boilerplate.

up to you though, I'll add the extra code if you don't find this convincing.

Reply:
Yea, I agree updating tests is unfortunate and not ideal. However, the other arguments are already using this pattern so I think keeping this consistent is better than mixing both approaches. If you'd like to explore making all of these arguments optional in a follow-up PR I would be very happy to review that.